genai-perf/genai_perf/inputs/retrievers/input_retriever_factory.py

```diff
@@ -68,57 +71,36 @@ def get_input_data(self) -> Dict:
             if self.config.output_format == OutputFormat.IMAGE_RETRIEVAL:
                 input_data = self._encode_images_in_input_dataset(input_data)
 
-            generic_dataset_json = (
-                self._convert_input_synthetic_or_file_dataset_to_generic_json(
-                    input_data
-                )
-            )
         else:
             if self.config.input_type == PromptSource.SYNTHETIC:
-                synthetic_dataset = self._get_input_dataset_from_synthetic()
-                generic_dataset_json = (
-                    self._convert_input_synthetic_or_file_dataset_to_generic_json(
-                        synthetic_dataset
-                    )
-                )
+                synthetic_retriever = SyntheticDataRetriever(self.config)
+                input_data = synthetic_retriever.retrieve_data()
+                # synthetic_dataset = self._get_input_dataset_from_synthetic()
             elif self.config.input_type == PromptSource.FILE:
                 # TODO: remove once the factory fully integrates retrievers
                 file_retriever = FileInputRetriever(self.config)
                 input_data = file_retriever.retrieve_data()
-
-                input_file_dataset = self._encode_images_in_input_dataset(input_data)
-                generic_dataset_json = (
-                    self._convert_input_synthetic_or_file_dataset_to_generic_json(
-                        input_file_dataset
-                    )
-                )
+                self._encode_images_in_input_dataset(input_data)
             else:
                 raise GenAIPerfException("Input source is not recognized.")
 
-        return generic_dataset_json
+        return input_data
 
     def _convert_input_synthetic_or_file_dataset_to_generic_json(
         self, dataset: Dict
     ) -> Dict[str, List[Dict]]:
+        raise GenAIPerfException("STOP")
```

Comment:
What is the purpose of this exception? It isn't descriptive and has no qualification.

Reply:
Whoops, this should not be in here. This was used during the initial debugging steps. I removed it. Thanks for catching this!